modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/metadata/DefaultComponentMetadataConfigurer.java

```diff
@@ -164,6 +173,9 @@ public DefaultComponentMetadataConfigurer setConnected(boolean connected) {
 
   @Override
   public DefaultComponentMetadataConfigurer asOneOfRouter() {
+    if (chainInputTypeResolver == null) {
```

Comment:
not a good default

Reply:
I know, we talked about this some time ago. I'm going to change it. Was just uploading the draft PR. Thanks for the reminder though.

Reply:
done